File: hideapi/src/main/java/$android/content/Context.java
package $android.content;

import android.content.BroadcastReceiver;
import android.content.Intent;
import android.content.IntentFilter;
import android.os.Handler;
import android.os.UserHandle;

import utils.Utils;

public class Context {
    public Intent registerReceiverAsUser(
            BroadcastReceiver receiver,
            UserHandle user,
            IntentFilter filter,
            String broadcastPermission,
            Handler scheduler
    ) {
        return Utils.throwStub();
    }
}

File: module/src/main/java/com/github/kr328/intent/compat/Context.kt
package com.github.kr328.intent.compat

import android.content.BroadcastReceiver
import android.content.Context
import android.content.ContextHidden
import android.content.IntentFilter
import android.os.Handler
import android.os.Looper
import android.os.UserHandle
import com.github.kr328.intent.util.unsafeCast

fun Context.registerReceiverAsUser(
    receiver: BroadcastReceiver,
    user: UserHandle,
    filter: IntentFilter,
    broadcastPermission: String?,
    scheduler: Handler = Handler(Looper.getMainLooper()),
) {
    unsafeCast<ContextHidden>().registerReceiverAsUser(
        receiver,
        user,
        filter,
        broadcastPermission,
        scheduler,
    )
}
